### PR TITLE
FLYTECTL_CONFIG env var should take highest precedence

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
Also when running `pyflyte -c cfg.yaml package ...`, if a user has `Config.auto()` in their code, `cfg.yaml` should get used.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: flytekit/clis/sdk_in_container/pyflyte.py [hunk 1]
```diff
@@ -120,6 +122,14 @@ def main(ctx, pkgs: typing.List[str], config: str, verbose: bool):
     if config:
         ctx.obj[CTX_CONFIG_FILE] = config
         cfg = configuration.ConfigFile(config)
+        # Set here so that if someone has Config.auto() in their user code, the config here will get used.
+        if FLYTECTL_CONFIG_ENV_VAR not in os.environ:
+            os.environ[FLYTECTL_CONFIG_ENV_VAR] = config
+        else:
+            cli_logger.warning(
+                f"Config file in arg {config} will be overridden by env var {FLYTECTL_CONFIG_ENV_VAR}:"
```
Comment (pingsutw) [inline flytekit/clis/sdk_in_container/pyflyte.py:130]: I feel like `args` should take higher precedence than `env`. here is the [doc](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html) for aws credentials precedence.

Comment (wild-endeavor) [reply flytekit/clis/sdk_in_container/pyflyte.py:130]: sure.

**Review by pingsutw — APPROVED** [review]
